header.go

```diff
@@ -24,9 +25,28 @@ import (
 // "Google-Cloud-").
 //
 // Currently, the following keys are reserved: Accept, Accept-Encoding,
-// Accept-Post, Allow, Content-Encoding, Content-Type, Te, and Trailer. Keys
-// prefixed with "Grpc-", "Rerpc-", and "Twirp-" are also reserved.
+// Accept-Post, Allow, Content-Encoding, Content-Type, Te, and Trailer. Empty
+// keys, or keys prefixed with ":", "Grpc-", "Rerpc-", and "Twirp-" are also
+// reserved.
+//
+// Unreserved keys can only contain the following ASCII characters:
+// a-z, A-Z, 0-9, "-" (hyphen-minus), "_" (underscore), and "." (period).
 func IsReservedHeader(key string) error {
+	if key == "" {
+		return errors.New("empty header key")
+	}
+	if key[0] == ':' {
+		return fmt.Errorf("%q is a reserved HTTP2 proto-header", key)
```

Comment:
Should this be "...HTTP2 *pseudo*-header"?

Reply:
Good catch!